activation/nipst.go

```diff
@@ -214,18 +215,19 @@ func (nb *NIPSTBuilder) BuildNIPST(challenge *types.Hash32, atxExpired, stop cha
 
 	// Phase 2: PoST execution.
 	if nipst.PostProof == nil {
-		nb.log.Info("starting PoST execution (challenge: %x)", nb.state.PoetProofRef)
-
+		nb.log.With().Info("starting PoST execution",
+			log.String("challenge", fmt.Sprintf("%x", nb.state.PoetProofRef)))
+		startTime := time.Now()
 		proof, err := nb.postProver.Execute(nb.state.PoetProofRef)
 		if err != nil {
 			return nil, fmt.Errorf("failed to execute PoST: %v", err)
 		}
 
-		//todo remove (#1923)
-		//this is just for debugging
 		b, _ := types.InterfaceToBytes(proof)
```

Comment:
We serialize the proof here just to print the length of the serialized string. This todo was to remove the extra serialization for performance reasons.

Three options:
1. Keep the todo
2. Remove the length of the proof from the log and remove the extra serialization (<-- my fav.)
3. Find a way to get this size without doing an extra serialization

I've noticed that you've already closed #1923. This is not good practice. You can link that issue to this PR and it will be closed when you close the PR. This will also add a comment to the issue automatically so people will know to look here before working on it.

Reply:
I made change 2 as you suggested.

I looked into this, and 3. would probably require changing the Proof type to track/cache its own size, or something of this sort, which seems too deep a change for a simple debug statement.